File: challenges/challenge-01/C01.ts
export function resolve() {
    const usersInput = Deno.readTextFileSync('users.txt');

    const users = usersInput
        .replace(/\r|\n/g, '~')
        .split('~~~~')
        .map(user => user.replace(/~~/g, ' '));
    
    const result = users.reduce((acc, user) => {
        const userProps = user.split(' ')
            .map(prop => ({
                key: prop.substring(0, 3),
                value: prop.substring(4)
            }));
        
        let usr = '';
        const validAttributes = new Set(['usr', 'eme', 'psw', 'age', 'loc', 'fll'])
        for (const { key, value } of userProps) {
            validAttributes.delete(key)
            if (key === 'usr') usr = value;
        }
        
        if (validAttributes.size === 0) {
            acc.valid_users += 1;
            acc.last_user = usr;
        }
        
        return acc;
    }, { valid_users: 0, last_user: '' })

    return `${result.valid_users}:${result.last_user}`;
}

File: challenges/challenge-01/C01.cs
public static class Challenge01
{
    public static string Resolve()
    {
        List<string[]> users = new List<string[]>();
        try
        {
            StreamReader sr = new StreamReader("users.txt");
            string? line = sr.ReadLine();

            string currentUser = "";
            while (line != null)
            {
                if (line != "")
                {
                    currentUser += $" {line}";
                }
                else
                {
                    var user = currentUser.Trim().Split(' ');
                    users.Add(user);
                    currentUser = "";
                }
                line = sr.ReadLine();
            }

            sr.Close();
        }
        catch (System.Exception e)
        {
            Console.WriteLine(e);
        }

        var result = (count: 0, lastUser: "");
        foreach (var userData in users)
        {
            Dictionary<string, int> validKeys = new Dictionary<string, int>() {
                { "usr", 0 },
                { "eme", 0 },
                { "psw", 0 },
                { "age", 0 },
                { "loc", 0 },
                { "fll", 0 }
            };
            
            string usr = "";
            foreach (var prop in userData)
            {
                var key = prop.Substring(0, 3);
                var value = prop.Substring(4);

                validKeys.Remove(key);

                if (key == "usr")
                {
                    usr = value;
                }
            }

            if (validKeys.Count == 0)
            {
                result.count += 1;
                result.lastUser = usr;
            }
        }

        return $"{result.count}:{result.lastUser}";
    }
}